src/components/svgs/stations/osaka-metro.tsx

```diff
@@ -0,0 +1,915 @@
+                            text={`(${oldName})`.split('\n')}
+                            fontSize={LAYOUT_CONSTANTS.FONT_SIZE.OLD_NAME}
+                            lineWidth={LAYOUT_CONSTANTS.FONT_SIZE.OLD_NAME}
+                            transform={`${`translate(${LAYOUT_CONSTANTS.FONT_SIZE.OLD_NAME - LAYOUT_CONSTANTS.FONT_SIZE.NAME - LAYOUT_CONSTANTS.MAGIC_OFFSET_8}, 0)`}${oldNameScale === 1 ? '' : ` scale(1, ${oldNameScale})`}`}
+                            grow="bidirectional"
+                            baseOffset={1}
+                            fontWeight="bold"
+                            {...getLangStyle(TextLanguage.tokyo_ja)}
+                        />
+                    )}
+                    <MultilineTextVertical
+                        text={names[1].split('\n')}
+                        fontSize={LAYOUT_CONSTANTS.FONT_SIZE.TRANSLATION}
+                        lineWidth={LAYOUT_CONSTANTS.FONT_SIZE.TRANSLATION}
+                        transform={`${`translate(${-(LAYOUT_CONSTANTS.FONT_SIZE.OLD_NAME + 2) * (hasOldName ? 1 : 0) - 4}, 0)`}${translationScale === 1 ? '' : ` scale(1, ${translationScale})`}`}
+                        grow="left"
+                        baseOffset={1}
+                        fontWeight="bold"
+                        {...getLangStyle(TextLanguage.tokyo_ja)}
+                    />
+                </g>
+            )}
+        </g>
+    );
+};
+
+const osakaMetroAttrsComponent = (props: AttrsProps<OsakaMetroStationAttributes>) => {
+    const { id, attrs, handleAttrsUpdate } = props;
+    const { t } = useTranslation();
+
+    const interchangeCount = attrs.transfer?.[0]?.length || 1;
+    const isHorizontal = attrs.nameDirection === 'horizontal';
+    const isMultipleTransfers = interchangeCount > 1;
+    const isNameUpOrDown = ['up', 'down'].includes(attrs.nameOverallPosition);
+
+    const handleAutoAdjustment = React.useCallback(() => {
+        if (isMultipleTransfers) {
+            let needsUpdate = false;
+            const newAttrs = { ...attrs };
+
+            if (newAttrs.stationType !== 'normal') {
+                newAttrs.stationType = 'normal';
+                needsUpdate = true;
+            }
+            if (newAttrs.nameDirection !== 'horizontal') {
+                newAttrs.nameDirection = 'horizontal';
+                needsUpdate = true;
+            }
+
+            if (needsUpdate) {
+                handleAttrsUpdate(id, newAttrs);
+            }
+        }
+    }, [isMultipleTransfers, attrs.stationType, attrs.nameDirection, id, handleAttrsUpdate]);
+
+    const syncOffsets = React.useCallback((updatedAttrs: OsakaMetroStationAttributes) => {
+        const { nameOffsetX, nameOffsetY } = convertToOffsetXY(
+            updatedAttrs.nameOverallPosition,
+            updatedAttrs.nameOffsetPosition
+        );
+        updatedAttrs.nameOffsetX = nameOffsetX;
+        updatedAttrs.nameOffsetY = nameOffsetY;
+        return updatedAttrs;
+    }, []);
```

Comment:
I can't find the reason to `useCallback` of this function. And since it is called only in `updateAttr`, you may directly run these code there. Or if you want to be wrapped in the function, move it to the root level as it is not capturing any closure.